aspnetcore/fundamentals/host/generic-host.md

```diff
@@ -1,31 +1,490 @@
+
+The `ConfigureWebHostDefaults` method:
+
+* Loads host configuration from environment variables prefixed with "ASPNETCORE_".
+* Sets [Kestrel](xref:fundamentals/servers/kestrel) server as the web server and configures it using the app's hosting configuration providers. For the Kestrel server's default options, see <xref:fundamentals/servers/kestrel#kestrel-options>.
+* Adds [Host Filtering middleware](xref:fundamentals/servers/kestrel#host-filtering).
+* Adds [Forwarded Headers middleware](xref:host-and-deploy/proxy-load-balancer#forwarded-headers) if ASPNETCORE_FORWARDEDHEADERS_ENABLED=true.
+* Enables IIS integration. For the IIS default options, see <xref:host-and-deploy/iis/index#iis-options>.
+
+The [Settings for all app types](#settings-for-all-app-types) and [Settings for web apps](#settings-for-web-apps) sections later in this article show how to override default builder settings.
+
+## Framework-provided services
+
+Services that are registered automatically include the following:
+
+* [IHostApplicationLifetime](#ihostapplicationlifetime)
+* [IHostLifetime](#ihostlifetime)
+* [IHostEnvironment / IWebHostEnvironment](#ihostenvironment)
+
+For a list of all framework-provided services, see <xref:fundamentals/dependency-injection#framework-provided-services>.
+
+## IHostApplicationLifetime
+
+Inject the <xref:Microsoft.Extensions.Hosting.IHostApplicationLifetime> (formerly `IApplicationLifetime`) service into any class to handle post-startup and graceful shutdown tasks. Three properties on the interface are cancellation tokens used to register app start and app stop event handler methods. The interface also includes a `StopApplication` method.
+
+The following example is an `IHostedService` implementation that registers the `IApplicationLifetime` events:
+
+[!code-csharp[](generic-host/samples-snapshot/3.x/LifetimeEventsHostedService.cs?name=snippet_LifetimeEvents)]
+
+## IHostLifetime
+
+The <xref:Microsoft.Extensions.Hosting.IHostLifetime> implementation controls when the host starts and when it stops. The last implementation registered is used.
+
+<xref:Microsoft.Extensions.Hosting.Internal.ConsoleLifetime> is the default `IHostLifetime` implementation. `ConsoleLifetime`:
+
+* listens for Ctrl+C/SIGINT or SIGTERM and calls <xref:Microsoft.Extensions.Hosting.IApplicationLifetime.StopApplication*> to start the shutdown process.
+* Unblocks extensions such as [RunAsync](#runasync) and [WaitForShutdownAsync](#waitforshutdownasync).
+
+## IHostEnvironment
+
+Inject the <xref:Microsoft.Extensions.Hosting.IHostEnvironment> service into a class to get information about the following:
+
+* [ApplicationName](#applicationname)
+* [EnvironmentName](#environmentname)
+* [ContentRootPath](#contentrootpath)
+
+Web apps implement the `IWebHostEnvironment` interface, which inherits `IHostEnvironment` and adds:
+
+* [WebRootPath](#webroot)
+
+## Host configuration
+
+Host configuration is used for the following:
```

Comment:
It's available from HostBuilderContext.Configuration inside ConfigureAppConfiguration. After ConfigureAppConfiguration HostBuilderContext.Configuration is replaced with the app config.
https://github.com/aspnet/Extensions/blob/36a9bf9d3fdea32b548d7001b27e6b19ae78b094/src/Hosting/Hosting/src/HostBuilder.cs#L169-L190